R/html_dependencies.R

```diff
@@ -194,16 +194,26 @@ html_reference_path <- function(path, lib_dir, output_dir) {
 # return the html dependencies as an HTML string suitable for inclusion
 # in the head of a document
 html_dependencies_as_string <- function(dependencies, lib_dir, output_dir) {
+  # Only copy deps with path
+  dependencies_href <- filter_dependencies(dependencies, "href")
+  dependencies <- filter_dependencies(dependencies, "file")
```

Comment:
I think it's not necessary to split the dependencies into `file` and `href` if you take advantage of two things:

1. `copyDependencyToDir` and `makeDependencyRelative` can both be passed `mustWork = FALSE` in which case it will just ignore non-file based dependencies
2. `renderDependencies` will take multiple src types, e.g. `srcType = c("file", "href")`, and it will use the first of those that's present in the dependency.

Reply:
> copyDependencyToDir and makeDependencyRelative can both be passed mustWork = FALSE in which case it will just ignore non-file based dependencies

That will simplify indeed ! thanks

> renderDependencies will take multiple src types, e.g. srcType = c("file", "href"), and it will use the first of those that's present in the dependency.

It seems we use special `encodeFunc = identity` and a special `hrefFilter=` function for `srcType = "file"`. 
Currently it prevents us to do one call for all srcType because we don't want `href` srcType to be processed with those functions. 
https://github.com/rstudio/rmarkdown/blob/145046175c721af185aa6ba3ecc9262f12dc7369/R/html_dependencies.R#L306-L309

It is an old code but it could have been a requirement for Pandoc. Processing both differently would lead to this https://github.com/rstudio/rmarkdown/pull/1948#discussion_r608469286

Do you remember maybe ? (long shot from blame finding https://github.com/rstudio/rmarkdown/commit/4ad77fa4b3782c8c3628ddb035474a5a03e18d66#diff-028517fb0fb57172622ddaa0acc9296ae94664b57e4d390760e14c0c835301c3R74)

Reply:
Ah, sorry, I have no recollection. If you need me to I can try to figure it out.